tests/foreman/cli/test_activationkey.py

```diff
@@ -1520,42 +1419,37 @@ def test_positive_view_subscriptions_by_non_admin_user(
                 Search: "name ~ "Test_*_Dev" || name ~ "Test_*_QA"
             Organization:
                 view_organizations, assign_organizations,
-            Katello::Subscription:
-                view_subscriptions, attach_subscriptions,
-                unattach_subscriptions
+            Katello::Product:
+                view_products, edit_products, sync_products,
+                create_products, destroy_products
 
 
     :expectedresults: The non admin user can view the activation key
-        subscription
+        repository sets
 
     :BZ: 1406076
     """
-    org = module_entitlement_manifest_org
+    org = module_sca_manifest_org
     user_name = gen_alphanumeric()
     user_password = gen_alphanumeric()
-    ak_name_like = f'ak_{gen_string("alpha")}'
     hc_names_like = (
         f'Test_*_{gen_string("alpha")}',
         f'Test_*_{gen_string("alpha")}',
     )
-    ak_name = f'{ak_name_like}_{gen_string("alpha")}'
-    available_subscriptions = module_target_sat.cli.Subscription.list(
-        {'organization-id': org.id}, per_page=False
-    )
-    assert len(available_subscriptions) > 0
-    available_subscription_ids = [subscription['id'] for subscription in available_subscriptions]
-    subscription_id = choice(available_subscription_ids)
-    activation_key = module_target_sat.cli_factory.make_activation_key(
-        {'name': ak_name, 'organization-id': org.id}
-    )
-    module_target_sat.cli.ActivationKey.add_subscription(
-        {'id': activation_key['id'], 'subscription-id': subscription_id}
+    result = module_target_sat.cli_factory.setup_org_for_a_rh_repo(
+        {
+            'product': PRDS['rhel'],
+            'repository-set': REPOSET['rhst7'],
+            'repository': REPOS['rhst7']['name'],
```

Comment:
I would prefer single source of truth, all you need should be defined in `REPOS` already.
```suggestion
            'product': REPOS['rhst7']['product'],
            'repository-set': REPOS['rhst7']['reposet'],
            'repository': REPOS['rhst7']['name'],
```
Newer `rhst8` might work too.